hack/e2e.sh

```diff
@@ -13,32 +88,19 @@ mkdir -p ${GOPATH}/src/k8s.io
 cd ${GOPATH}/src/k8s.io && git clone -b ${RELEASE_VERSION} --single-branch https://${AUTOSCALER_PKG}.git autoscaler
 
 echo "Check the VerticalPodAutoScalerController configurations ..."
+SCRIPT_ROOT=${GOPATH}/src/k8s.io/autoscaler/vertical-pod-autoscaler/
+
 recommendationOnly=$(kubectl get VerticalPodAutoScalerController default -n openshift-vertical-pod-autoscaler -oyaml|yq ".spec.recommendationOnly" || false)
+run_upstream_vpa_tests
 
-SCRIPT_ROOT=${GOPATH}/src/k8s.io/autoscaler/vertical-pod-autoscaler/
-if $recommendationOnly
+kubectl patch verticalpodautoscalercontroller default -n openshift-vertical-pod-autoscaler --type merge --patch '{"spec":{"recommendationOnly": false}}'
+curstatus=$(await_for_controllers "$WAIT_TIME")
+if [[ "$curstatus" == "all" ]];
 then
-  echo "recommendationOnly is enabled. Run the recommender e2e tests in upstream ..."
-  pushd ${SCRIPT_ROOT}/e2e
-  GO111MODULE=on go test -mod vendor ./v1/*go -v --test.timeout=60m --args --ginkgo.v=true --ginkgo.focus="\[VPA\] \[recommender\]" --ginkgo.skip="doesn't drop lower/upper after recommender's restart" --report-dir=/workspace/_artifacts --disable-log-dump
-  V1_RESULT=$?
-  popd
-  echo "v1 recommender test result:" ${V1_RESULT}
-
-  if [ $V1_RESULT -gt 0 ]; then
-    echo "Tests failed"
-    exit 1
-  fi
+  echo "All controllers are running"
 else
-  echo "recommendationOnly is disabled. Run the full-vpa e2e tests in upstream ..."
-  pushd ${SCRIPT_ROOT}/e2e
-  GO111MODULE=on go test -mod vendor ./v1/*go -v --test.timeout=60m --args --ginkgo.v=true --ginkgo.focus="\[VPA\] \[full-vpa\]" --report-dir=/workspace/_artifacts --disable-log-dump
-  V1_RESULT=$?
-  popd
-  echo "v1 full-vpa test result:" ${V1_RESULT}
-
-  if [ $V1_RESULT -gt 0 ]; then
-    echo "Tests failed"
-    exit 1
-  fi
-fi
\ No newline at end of file
+  echo "error - not all controllers are running!"
+  exit 1
+fi
+recommendationOnly=$(kubectl get VerticalPodAutoScalerController default -n openshift-vertical-pod-autoscaler -oyaml|yq ".spec.recommendationOnly" || false)
```

Comment:
In the case that `kubectl` fails, do you want the script to exit? Or is the `|| false` supposed to put `false` into `recommendationOnly`?

If you want the script to exit, I think you can simply remove `|| false` since it will already exit due to the `-e` shell setting. With or without `|| false` it will exit.

If you wanted a `false` value for `recommendationOnly`, then the end should probably be `|| echo false` instead of `|| false`. That would cause the string `false` to be in `recommendationOnly` if the `kubectl` fails, and the script would continue running.

Reply:
> In the case that `kubectl` fails, do you want the script to exit? Or is the `|| false` supposed to put `false` into `recommendationOnly`?
> 
> If you want the script to exit, I think you can simply remove `|| false` since it will already exit due to the `-e` shell setting. With or without `|| false` it will exit.
> 
> If you wanted a `false` value for `recommendationOnly`, then the end should probably be `|| echo false` instead of `|| false`. That would cause the string `false` to be in `recommendationOnly` if the `kubectl` fails, and the script would continue running.

I wanted a false value for recommendationOnly, then the end should probably be || echo false. I will change accordingly.